### DISCUSSION TITLE
TextBox  Low performance

### DISCUSSION THREAD

**Discussion by chenzanyu** [AvaloniaUI/Avalonia#12102] [Q&A] [answered]
<TextBox Name="testTextBox" Margin="10" HorizontalAlignment="Stretch" VerticalAlignment="Stretch">
		
    </TextBox>

**Comment by chenzanyu** [comment]
```
public partial class MainView : UserControl
    {
        public MainView()
        {
            InitializeComponent();
            SetTestTextBoxText();
        }

        private void SetTestTextBoxText()
        {
            testTextBox.Text = string.Join("\r\n", Enumerable.Range(0, 20000).Select(x => x.ToString()));
        }
    }

```

**Comment by chenzanyu** [comment]
When the textbox content is more content, enter, select the text function stuck(use Avalonia version:11.0)

**Comment by BAndysc** [comment]
When you want to display a text field with a lot of lines, you should rather use https://github.com/AvaloniaUI/AvaloniaEdit which renders only visible lines thus can handle large text

**Reply by chenzanyu** [reply]
I’ve noticed that lagging is not related to the number of lines. If the string length reaches 5000, there is significant lag. I’m inclined to believe that this is a bug rather than an intentional design

**Comment by hez2010** [comment, marked as answer]
If you think this is a bug, I recommend you create an issue for this so that the team can track it.

**Comment by chenzanyu** [comment]
#12108
